src/lib.rs

```diff
@@ -22,8 +59,50 @@ use std::fs;
 use std::path::{self, PathBuf, Path};
 use rand::{thread_rng, Rng};
 
-/// A wrapper for a path to temporary directory implementing automatic
-/// scope-based deletion.
+/// A directory in the filesystem that is automatically deleted when
+/// it goes out of scope.
+///
+/// The [`TempDir`] type creates a directory on the file system that
+/// is deleted once it goes out of scope. At construction, the
+/// `TempDir` creates a new directory with a randomly generated name,
+/// and with a prefix of your choosing.
+///
+/// The default constructor, [`TempDir::new`], creates directories in
+/// the location returned by [`std::env::temp_dir()`], but they can be
+/// configured to manage the lifetime of a temporary directory in any
```

Comment:
Perhaps "lifetime" is not the best choice of word here given its significance in Rust.